LinkedLists/addLists/solution1.js

```diff
@@ -0,0 +1,92 @@
+// Given two linked lists. Each node contains a single digit. 
+// The digits are stored in reverse order, 
+// such that the 1st digit is at the head of the list. 
+// Write a function that adds the two numbers and returns the sum as a linked list.
+
+// creates a node
+function ListNode(value, next) {
+    this.value = value;
+    this.next = next;
+}
+
+// helper function to print all values in the node
+function printValues(head) {
+    let array = [];
+    let temp = head;
+    while (temp != null) {
+        array.push(temp.value);
+        temp = temp.next;
+    }
+    console.log("ARRAY", array);
+}
+
+// Time complexity is Big O(N);
+// Space complexity is Big O(N) because are making a stack with recursive calls;
+function addLists(l1, l2) {
```

Comment:
@annamikmorozova I'm not understanding why 'carry' doesn't have to be passed into addLists?

Reply:
We could, but we do not need to know the internals, but only need to know the linked lists. That's why I made a separate function that would handle "carry"

Reply:
we would have to pass in "carry" if we would have it all in on function, but we have a separate function to handle it

Reply:
but carry isn't defined in add lists anywhere, so why does the carry == 0 check work? Doesn't it just return true because carry is not defined because you didn't do a === check? So it is returning falsy even though it seems like the code is supposed to check if the addListsHelper passes 0 in for carry?

Reply:
seems like you expect the 3 para of addLists to be auto passed? but I don't think it is defined by the time it gets to 26? I might be wrong, but that's what it looks like to me

Reply:
I did not even notice! Thank you for seeing this error!! I will correct now

Reply:
fixed! Thank you again for the review!